src/Workspaces/Core/Portable/Shared/Extensions/SyntaxEditorExtensions.cs

```diff
@@ -13,6 +13,29 @@ namespace Microsoft.CodeAnalysis.Shared.Extensions
 {
     internal static class SyntaxEditorExtensions
     {
+        /// <summary>
+        /// Performs several edits to a document.  If multiple edits are made within the same
+        /// expression context, then the document/semantic-model will be forked after each edit 
+        /// so that further edits can see if they're still safe to apply.
+        /// </summary>
+        public static Task ApplyExpressionLevelSemanticEditsAsync<TType>(
+            this SyntaxEditor editor, Document document,
+            ImmutableArray<TType> originalNodes,
+            Func<SemanticModel, TType, SyntaxNode, bool> canReplace,
+            Func<SemanticModel, SyntaxNode, TType, SyntaxNode, SyntaxNode> updateRoot,
+            Func<TType, SyntaxNode> selector,
+            CancellationToken cancellationToken)
+        {
+            return ApplySemanticEditsAsync(
+                editor, document,
+                originalNodes,
+                (syntaxFacts, node) => GetExpressionSemanticBoundary(syntaxFacts, node),
+                canReplace,
+                updateRoot,
+                selector,
+                cancellationToken);
+        }
```

Comment:
i woudl expect the existing ApplyExpressionLevelSemanticEditsAsync to call into this more general helper, passing along `node => node` as the selector function.